src/ansys/fluent/core/session.py

```diff
@@ -368,12 +307,261 @@ def _exit(
         if remote_instance:
             remote_instance.delete()
 
+
+class BaseSession:
+    """Instantiates a Fluent connection.
+
+    Attributes
+    ----------
+    scheme_eval: SchemeEval
+        Instance of SchemeEval on which Fluent's scheme code can be
+        executed.
+
+    Methods
+    -------
+    create_from_server_info_file(
+        server_info_filepath, cleanup_on_exit, start_transcript
+        )
+        Create a Session instance from server-info file
+
+    start_transcript()
+        Start streaming of Fluent transcript
+
+    stop_transcript()
+        Stop streaming of Fluent transcript
+
+    check_health()
+        Check health of Fluent connection
+
+    exit()
+        Close the Fluent connection and exit Fluent.
+    """
+
+    def __init__(
+        self,
+        ip: str = None,
+        port: int = None,
+        password: str = None,
+        channel: grpc.Channel = None,
+        cleanup_on_exit: bool = True,
+        start_transcript: bool = True,
+        remote_instance=None,
+        fluent_connection=None,
+    ):
+        if not fluent_connection:
+            self.fluent_connection = _FluentConnection(
+                ip=ip,
+                port=port,
+                password=password,
+                channel=channel,
+                cleanup_on_exit=cleanup_on_exit,
+                start_transcript=start_transcript,
+                remote_instance=remote_instance,
+            )
+        else:
+            self.fluent_connection = fluent_connection
+
+        self.scheme_eval = self.fluent_connection.scheme_eval
+
+    @classmethod
+    def create_from_server_info_file(
+        cls,
+        server_info_filepath: str,
+        cleanup_on_exit: bool = True,
+        start_transcript: bool = True,
+    ) -> "BaseSession":
+        """Create a Session instance from server-info file.
+
+        Parameters
+        ----------
+        server_info_filepath : str
+            Path to server-info file written out by Fluent server
+        cleanup_on_exit : bool, optional
+            When True, the connected Fluent session will be shut down
+            when PyFluent is exited or exit() is called on the session
+            instance, by default True.
+        start_transcript : bool, optional
+            The Fluent transcript is started in the client only when
+            start_transcript is True. It can be started and stopped
+            subsequently via method calls on the Session object.
+            Defaults to true.
+
+        Returns
+        -------
+        Session
+            Session instance
+        """
+        ip, port, password = parse_server_info_file(server_info_filepath)
+        session = BaseSession(
```

Comment:
@prmukherj related to my comment in session_meshing.py, is it the case that it's immediately possible simply to remove all the other `create_from_server_info_file` implementations and meanwhile change `BaseSession` to `cls` in the above line?